.github/workflows/lint-and-test.yml

```diff
@@ -0,0 +1,69 @@
+name: Linting & tests
+
+on: [push, pull_request]
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    name: Lint - Type consistency (mypy)
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install with linting tools
+        run: pip install .[linting]
+      - name: Run mypy
+        run: ./tools/run-mypy
+
+  flake8:
+    runs-on: ubuntu-latest
+    name: Lint - PEP8 & more (flake8)
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install with linting tools
+        run: pip install .[linting]
+      - name: Run flake8
+        run: flake8
+
+  isort:
+    runs-on: ubuntu-latest
+    name: Lint - Import order (isort)
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.5
+      - name: Install with linting tools
+        # NOTE: Install pytest so that isort recognizes it as a known library
+        run: pip install .[linting] && pip install pytest
+      - name: Run isort
+        run: ./tools/run-isort-check
+
+  pytest:
+    runs-on: ubuntu-latest
+    strategy:
+      # Not failing fast allows all matrix jobs to try & finish even if one fails early
+      fail-fast: false
+      matrix:
+        test-python-version: [3.5, 3.6, 3.7, 3.8, 3.9]
+    name: Install & pytest - CPython ${{ matrix.test-python-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Python 3
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.test-python-version }}
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip
+      - name: Ensure regular package installs from checkout
+        run: pip install .
+      - name: Install test dependencies
+        run: pip install .[testing]
+      - name: Run tests with pytest
+        run: pytest 
+      - name: Run codecov
```

Comment:
Should we run `codecov` in every python version? We could consider checking a conditional here and running it on one python version.

Reply:
Good question - I think it's run in every version currently, but I'm not sure if we should simplify this. Feel free to investigate!

Reply:
This was resolved in #869.